File: solutions/1512. Number of Good Pairs/1512.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int numIdenticalPairs(const vector<int>& nums) {
    int ans = 0;
    vector<int> count(101);

    for (const int num : nums)
      ans += count[num]++;

    return ans;
  }
};

File: solutions/1512. Number of Good Pairs/1512.rs
// Time complexity: O(n)
// Space complexity: O(1)
impl Solution {
    pub fn num_identical_pairs(nums: Vec<i32>) -> i32 {
        let mut ans = 0;
        let mut count = vec![0; 101];

        for num in nums {
            ans += count[num as usize];
            count[num as usize] += 1;
        }

        ans
    }
}
